src/module.ts

```diff
@@ -1,68 +1,64 @@
 import { fileURLToPath } from 'url'
 import fsp from 'fs/promises'
-import { defineNuxtModule, addPlugin, resolveModule, addServerMiddleware } from '@nuxt/kit'
+import { defineNuxtModule, resolveModule, createResolver } from '@nuxt/kit'
 import { parse, compileScript, compileTemplate } from '@vue/compiler-sfc'
 import type { SFCDescriptor } from '@vue/compiler-sfc'
-import type { Nitro } from 'nitropack'
 
-export interface ModuleOptions {
-  addPlugin: boolean
-}
+export interface ModuleOptions {}
 
 export default defineNuxtModule<ModuleOptions>({
   meta: {
     name: 'nuxt-component-meta',
     configKey: 'componentMeta'
   },
-  defaults: {
-    addPlugin: true
-  },
-  setup (options, nuxt) {
-    if (options.addPlugin) {
-      const runtimeDir = fileURLToPath(new URL('./runtime', import.meta.url))
-      let nitro: Nitro
-      nuxt.hook('components:extend', async (components) => {
-        const componentMeta = await Promise.all(
-          components.map(async (component) => {
-            const name = (component as any).pascalName
-            const path = resolveModule((component as any).filePath, { paths: nuxt.options.rootDir })
-            const source = await fsp.readFile(path, { encoding: 'utf-8' })
-
-            // Parse component source
-            const { descriptor } = parse(source)
-
-            // Parse script
-            const { props } = descriptor.scriptSetup
-              ? parseSetupScript(name, descriptor)
-              : {
-                  props: []
-                }
-
-            const { slots } = parseTemplate(name, descriptor)
-
-            return {
-              name,
-              props,
-              slots
-            }
-          })
-        )
-        nitro.options.virtual['#meta/virtual/meta'] = `export const components = ${JSON.stringify(componentMeta)}`
```

Comment:
nitro shoulnd't be available here if there wasn't the hack. Modifications to `nitro.options` after `nitro:init` are not picked and also nitro instance for prerendering, uses fresh clone of initial options. This virtual wasn't available because of this.